### DISCUSSION TITLE
[BoxPlot] How to create an offset with time based X axis

### DISCUSSION THREAD

**Discussion by LoiKos** [airbnb/visx#1063] [Q&A] [answered]
I'm working on a boxplot graph and i have some difficulties with xAxis : 

<img width="1159" alt="Capture d’écran 2021-02-11 à 18 23 40" src="https://user-images.githubusercontent.com/9548473/107673415-45fc7800-6c96-11eb-95fd-ee003c03c527.png">

```js
  const xValues = props.fakeData.data.map((item) => item.boxPlot.x)
  const xScaleDomain = getMinMax(xValues)
  const xScale = scaleUtc({
    domain: xScaleDomain,
    range: [0, width],
    clamp: true,
  })
```

```
[Mon Feb 01 2021 00:00:00 GMT+0100 (heure normale d’Europe centrale), Fri Jan 01 2021 00:00:00 GMT+0100 (heure normale d’Europe centrale), Tue Dec 01 2020 00:00:00 GMT+0100 (heure normale d’Europe centrale), Sun Nov 01 2020 00:00:00 GMT+0100 (heure normale d’Europe centrale), Thu Oct 01 2020 00:00:00 GMT+0200 (heure d’été d’Europe centrale), Tue Sep 01 2020 00:00:00 GMT+0200 (heure d’été d’Europe centrale), Sat Aug 01 2020 00:00:00 GMT+0200 (heure d’été d’Europe centrale), Wed Jul 01 2020 00:00:00 GMT+0200 (heure d’été d’Europe centrale), Mon Jun 01 2020 00:00:00 GMT+0200 (heure d’été d’Europe centrale), Fri May 01 2020 00:00:00 GMT+0200 (heure d’été d’Europe centrale), Wed Apr 01 2020 00:00:00 GMT+0200 (heure d’été d’Europe centrale), Sun Mar 01 2020 00:00:00 GMT+0100 (heure normale d’Europe centrale)]0: Mon Feb 01 2021 00:00:00 GMT+0100 (heure normale d’Europe centrale) {}1: Fri Jan 01 2021 00:00:00 GMT+0100 (heure normale d’Europe centrale) {}2: Tue Dec 01 2020 00:00:00 GMT+0100 (heure normale d’Europe centrale) {}3: Sun Nov 01 2020 00:00:00 GMT+0100 (heure normale d’Europe centrale) {}4: Thu Oct 01 2020 00:00:00 GMT+0200 (heure d’été d’Europe centrale) {}5: Tue Sep 01 2020 00:00:00 GMT+0200 (heure d’été d’Europe centrale) {}6: Sat Aug 01 2020 00:00:00 GMT+0200 (heure d’été d’Europe centrale) {}7: Wed Jul 01 2020 00:00:00 GMT+0200 (heure d’été d’Europe centrale) {}8: Mon Jun 01 2020 00:00:00 GMT+0200 (heure d’été d’Europe centrale) {}9: Fri May 01 2020 00:00:00 GMT+0200 (heure d’été d’Europe centrale) {}10: Wed Apr 01 2020 00:00:00 GMT+0200 (heure d’été d’Europe centrale) {}11: Sun Mar 01 2020 00:00:00 GMT+0100 (heure normale d’Europe centrale)]
```

Does some know i can make something like padding that my graph stay in the bounds ?

**Comment by williaster** [comment, marked as answer]
Hey @LoiKos 👋  thanks for checking out `visx`. I think the answer to this depends on whether you need a `scaleUtc` (time) scale, or if you would consider a `band` scale which will evenly space `string`s representing your dates. 

Band scales support a `padding` which will leave space on the sides for you (see this discussion for more #1035). Note that this is the type of scale we use in our [`BoxPlot` demo](https://airbnb.io/visx/statsplot).

If you need a time scale, you will have to manually add a margin on the left/right of the chart so that your labels don't get clipped (content in SVGs gets clipped by HTML parents, which is why your labels are cutoff). Looks like you're already using a bottom margin to accommodate the x-axis, so you'd just need to do the same for the left/right (check [this demo](https://airbnb.io/visx/barstackhorizontal) for a real example).

**Reply by LoiKos** [reply]
Thanks ! I've found a way to handle this. For those who want to know how i handle it, i paste below my "padding" : 
```js
  const xValues = data.map((item) => item.boxPlot.x)
  xValues.sort((a, b) => a - b)
  const xfirst = xValues[0].clone().subtract(20, 'day')
  const xlast = xValues[xValues.length - 1].clone().add(20, 'day')
  const xScale = scaleTime({
    domain: [xfirst, xlast],
    range: [0, width],
  })
```
I use [dayjs](https://github.com/iamkun/dayjs) to handle Date but i believe you can do the same with other libs
